geomesa-spark-sql/geomesa-spark-sql-core/src/main/scala/org/locationtech/geomesa/spark/GeoMesaSpark.scala

```diff
@@ -0,0 +1,56 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.spark
+
+import java.util.ServiceLoader
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.geotools.data.Query
+import org.opengis.feature.simple.SimpleFeature
+
+trait SpatialRDDProvider {
+
+  def canProcess(params: java.util.Map[String, java.io.Serializable]): Boolean
+
+  def rdd(conf: Configuration,
+          sc: SparkContext,
+          dsParams: Map[String, String],
+          query: Query,
+          numberOfSplits: Option[Int]): RDD[SimpleFeature]
+
+  def rdd(conf: Configuration,
+          sc: SparkContext,
+          dsParams: Map[String, String],
+          query: Query,
+          useMock: Boolean = false,
```

Comment:
we can probably just pass mock in the datastore params and not require this separate function